src/object/srv_obj_migrate.c

```diff
@@ -3150,7 +3150,10 @@ ds_migrate_stop(struct ds_pool *pool, unsigned int version, unsigned int generat
 	uuid_copy(arg.pool_uuid, pool->sp_uuid);
 	arg.version = version;
 	arg.generation = generation;
-	rc = dss_thread_collective(migrate_fini_one_ult, &arg, 0);
+
+	rc = ds_pool_thread_collective(pool->sp_uuid,
+				       PO_COMP_ST_NEW | PO_COMP_ST_DOWN | PO_COMP_ST_DOWNOUT,
+				       migrate_fini_one_ult, &arg, 0);
```

Comment:
[Request for changes] The "DAOS-15145 pool: add pool collective function #13764" commit is buggy, and requires at least (I can't find any other myself) the following fixes:

- DAOS-15672 rebuild: Fix pool destroy hangs (#14183)
- DAOS-15514 container: fix container destroy failure (#14108)

Reply:
It was recommended by Niu as something that might help with https://daosio.atlassian.net/browse/DAOS-15386

Reply:
sorry, this was meant as a reply to the other comment.